### ISSUE TITLE
Kraus noise crashes MPS simulator

### ISSUE THREAD

**Issue by chriseclectic** [Qiskit/qiskit-aer#991]
### Informations

- **Qiskit Aer version**: 0.7
- **Python version**: 3.7
- **Operating system**: MacOS

### What is the current behavior?

While running some tests I was getting crashes when I tried to use Kraus noise on the MPS simulator.

### Steps to reproduce the problem

Run the following code:

```python
from qiskit import execute
from qiskit.circuit.library import QFT
from qiskit.providers.aer import QasmSimulator
from qiskit.providers.aer.noise import NoiseModel, amplitude_damping_error

error1 = amplitude_damping_error(0.2)
noise_model = NoiseModel()
noise_model.add_all_qubit_quantum_error(error1, ['id', 'u1', 'u2', 'u3'])
noise_model.add_all_qubit_quantum_error(error1.tensor(error1), ['cx'])

backend = QasmSimulator(method='matrix_product_state', noise_model=noise_model)

job = execute(QFT(5), backend)
```

This crashes the python kernel with a blas error:
```
ldb must be >= MAX(K,1): ldb=0 K=0. BLAS error: Parameter number 11 passed to cblas_zgemm had an invalid value
```

### What is the expected behavior?

No crash

### Suggested solutions

The error looks like something is going wrong with a matrix multiplication somewhere in the source code

**Comment by chriseclectic** [comment]
Closed by #1003

**Comment by chriseclectic** [comment]
@merav-aharoni looks like this is still an issue on master. A couple of times when running the above script I get a malloc error instead of the BLAS error:

```
python(56590,0x7000157e6000) malloc: *** error for object 0x7fa91b675a60: pointer being freed was not allocated
python(56590,0x7000187f8000) malloc: Incorrect checksum for freed object 0x7fa91b6ce188: probably modified after being freed.
Corrupt value: 0x20
python(56590,0x7000157e6000) malloc: *** set a breakpoint in malloc_error_break to debug
```

But if i run it several times in a row it seems to return to the BLAS error

**Comment by chriseclectic** [comment]
Hopefully fixed this time by #1188
